packages/dev/addons/src/navigation/types.ts

```diff
@@ -0,0 +1,273 @@
+            filter?: QueryFilter;
+            halfExtents?: IVector3Like;
+        }
+    ): void;
+
+    createNavMesh(meshes: Array<Mesh>, parameters: INavMeshParametersV2): CreateNavMeshresult;
+    createNavMeshAsync(meshes: Array<Mesh>, parameters: INavMeshParametersV2, completion?: (navmeshData: Uint8Array) => void): Promise<CreateNavMeshresult>;
+
+    computePathSmooth(
+        start: Vector3,
+        end: Vector3,
+        options?: {
+            filter?: QueryFilter;
+            halfExtents?: IVector3Like;
+
+            /**
+             * @default 256
+             */
+            maxPathPolys?: number;
+
+            /**
+             * @default 2048
+             */
+            maxSmoothPathPoints?: number;
+
+            /**
+             * @default 0.5
+             */
+            stepSize?: number;
+
+            /**
+             * @default 0.01
+             */
+            slop?: number;
+        }
+    ): Vector3[];
+}
+
+export type SteerTargetResult =
+    | {
+          /**
+           * Indicates whether the steering target computation was successful.
+           */
+          success: false;
+      }
+    | {
+          /**
+           * Indicates whether the steering target computation was successful.
+           */
+          success: true;
+          /**
+           * The position to steer towards.
+           */
+          steerPos: Vector3;
+          /**
+           * The flag indicating the type of steering position.
+           */
+          steerPosFlag: number;
+          /**
+           * The reference to the polygon that the steering position is associated with.
+           */
+          steerPosRef: number;
+          /**
+           * The points that make up the path to the steering position.
+           */
+          points: Vector3[];
+      };
+
+export const ComputePathError = {
+    START_NEAREST_POLY_FAILED: "START_NEAREST_POLY_FAILED",
+    END_NEAREST_POLY_FAILED: "END_NEAREST_POLY_FAILED",
+    FIND_PATH_FAILED: "FIND_PATH_FAILED",
+    NO_POLYGON_PATH_FOUND: "NO_POLYGON_PATH_FOUND",
+    NO_CLOSEST_POINT_ON_LAST_POLYGON_FOUND: "NO_CLOSEST_POINT_ON_LAST_POLYGON_FOUND",
+};
+
+export type ComputeSmoothPathErrorType = (typeof ComputePathError)[keyof typeof ComputePathError];
+
+export type ComputeSmoothPathResult = {
+    /**
+     * Indicates whether the path computation was successful.
+     */
+    success: boolean;
+    /**
+     * The error message if the path computation failed.
+     */
+    error?: {
+        /**
+         * The type of error that occurred during path computation.
+         * @remarks This will be one of the values from `ComputePathError`.
+         */
+        type: ComputeSmoothPathErrorType;
+        /**
+         * Statusring describing the error.
+         */
+        status?: number;
+    };
+    /**
+     * The computed path as an array of Vector3 points.
+     */
+    // TODO: IVector3Like instead of Vector3?
+    path: Vector3[];
```

Comment:
Would there be a path if there is also an error? Maybe consider making this a discriminated union to be more clear, perhaps something like:
```ts
export type ComputeSmoothPathResult = {
    success: true,
    path: Vector3[],
} | {
    success: false,
    error: { ... },
}
```
This way from a typing standpoint, path only exists for a successful result and error only exists for a failed result.

Reply:
It returns an empty array when it fails.